script/split_tests.py

```diff
@@ -0,0 +1,197 @@
+#!/usr/bin/env python3
+"""Helper script to split test into n buckets."""
+
+from __future__ import annotations
+
+import argparse
+from dataclasses import dataclass, field
+from math import ceil
+import os
+import subprocess
+import sys
+
+
+class Bucket:
+    """Class to hold bucket."""
+
+    def __init__(
+        self,
+    ):
+        """Initialize bucket."""
+        self.tests = 0
+        self._paths = []
+
+    def add(self, part: TestFolder | TestFile):
+        """Add tests to bucket."""
+        self.tests += part.total_tests
+        self._paths.append(part.path)
+
+    def get_paths_line(self) -> str:
+        """Return paths."""
+        return " ".join(self._paths) + "\n"
+
+
+class BucketHolder:
+    """Class to hold buckets."""
+
+    def __init__(self, tests_per_bucket: int, bucket_count: int) -> None:
+        """Initialize bucket holder."""
+        self._tests_per_bucket = tests_per_bucket
+        self._bucket_count = bucket_count
+        self._current_bucket = Bucket()
+        self._buckets: list[Bucket] = [self._current_bucket]
+        self._last_bucket = False
+
+    def split_tests(self, tests: TestFolder | TestFile) -> None:
+        """Split tests into buckets."""
+        if (
+            self._current_bucket.tests + tests.total_tests < self._tests_per_bucket
+        ) or self._last_bucket:
```

Comment:
Why just dump everything in the last bucket if all others are full? That means that if we don't constantly maintain the bucket parameters as Home Assistant grows, we might end up with 1 test runner running 2x as many tests as the other buckets.